runtimes/eden/src/migrations.rs

```diff
@@ -100,27 +143,63 @@ impl OnRuntimeUpgrade for MoveValidatorsSetToInvulnerables {
 			|| migration::have_storage_value(VALIDATORS_SET_MODULE, PRIME_ITEM, EMPTY_HASH)
 			|| migration::have_storage_value(POA_MODULE, STORAGE_VERSION_ITEM, EMPTY_HASH)
 		{
-			Err("Failed to remove ValidatorsSet and/or Poa")
-		} else {
-			let invulnerables = pallet_collator_selection::pallet::Pallet::<Runtime>::invulnerables();
-			log::info!(
-				target: TRY_RUNTIME,
-				"CollatorSelection::Invulnerables are {:?}",
-				invulnerables
-			);
-			let candidacy_bond = pallet_collator_selection::pallet::Pallet::<Runtime>::candidacy_bond();
-			log::info!(
-				target: TRY_RUNTIME,
-				"CollatorSelection::candidacy_bond is {:?}",
-				candidacy_bond
-			);
-			let validators: BoundedVec<AccountId, MaxMembers> = Decode::decode(&mut state.as_slice()).unwrap();
-			if invulnerables == validators && candidacy_bond == CANDIDACY_BOND {
-				log::info!(target: TRY_RUNTIME, "MoveValidatorsSetToInvulnerables was successful");
-				Ok(())
-			} else {
-				Err("CollatorSelection::Invulnerables are not the same as ValidatorsSet::Members")
+			return Err("Failed to remove ValidatorsSet and/or Poa");
+		}
+
+		let invulnerables = pallet_collator_selection::pallet::Pallet::<Runtime>::invulnerables();
+		log::info!(
+			target: TRY_RUNTIME,
+			"CollatorSelection::Invulnerables are {:?}",
+			invulnerables
+		);
+		let candidacy_bond = pallet_collator_selection::pallet::Pallet::<Runtime>::candidacy_bond();
+		log::info!(
+			target: TRY_RUNTIME,
+			"CollatorSelection::candidacy_bond is {:?}",
+			candidacy_bond
+		);
+		let validators: BoundedVec<AccountId, MaxMembers> = Decode::decode(&mut state.as_slice()).unwrap();
```

Comment:
could this unwrap be a `?`?

Reply:
Yes and it's actual better. Thank you for seeing it.